client/client.go

```diff
@@ -777,8 +777,7 @@ func (r *NotaryRepository) saveMetadata(ignoreSnapshot bool) error {
 	}
 
 	for role, blob := range targetsToSave {
-		parentDir := filepath.Dir(role)
-		os.MkdirAll(parentDir, 0755)
+		// If the parent directory does not exist, the cache.Set will create it
```

Comment:
@HuKeping: yup this function is only used for clients saving local copies of the repo metadata to files on disk